ext/ffi/lib.rs

```diff
@@ -1077,6 +1189,19 @@ where
       NativeType::Pointer | NativeType::Function => NativeValue {
         pointer: cif.call::<*const u8>(*fun_ptr, &call_args),
       },
+      NativeType::Struct(fields) => {
+        let size = NativeType::compute_struct_size(fields);
+        let mut result = vec![0u8; size];
```

Comment:
I would go with a boxed slice. However, creating the uninit boxed slice seems to be a bit of a pain. There's this nightly feature https://doc.rust-lang.org/std/boxed/struct.Box.html#method.new_uninit_slice which would do exactly what we would want but without that feature it's a tough operation.

Something like this:
```rs
let ptr = if size == 0 {
  NonNull::dangling()
} else {
  // Calculate the size of the layout
  let layout =
    NonZeroLayout::array::<mem::MaybeUninit<u8>>(size).expect("capacity overflow");
    std::mem::Global.alloc(layout)?
};
let box: Box<[MaybeUninit<u8>; size] = unsafe {
  let slice = slice::from_raw_parts_mut(ptr.cast().as_ptr(), size);
  Box::from_raw(slice.as_ptr())
};
```

Then, converting the `Box<[MaybeUninit<u8>; size]>` into a `Box<[u8]>` would again be easy with this nightly API: https://doc.rust-lang.org/std/boxed/struct.Box.html#method.assume_init-1
Without it I think we need to use `std::mem::transmute`.

Reply:
Hmm. I Guess something like
```rs
// SAFETY: `[T; size]` is a valid array even for `MaybeUninit<U>` as `T`
let box: Box<[MaybeUninit<u8>; size]> = Box::new(unsafe { MaybeUninit<[MaybeUninit<u8>; size]>::uninit().assume_init() });
```
would also work. This is again from the nightly code's sources.

Reply:
MaybeUninit is stable.

Reply:
We can only put constant values in generic. `size` in your code must be compile time constant. IMO we should just stick with current way...

Reply:
You're very much correct. It might still be possible to use a `MaybeUninit<u8>` as the vector content but `.into_boxed_slice()` seems to be the proper thing here.